frontend/src/lib/components/room/Chat.svelte

```diff
@@ -26,6 +30,15 @@
       return acc
     }, []),
   )
+  const typingUsers = $derived(
+    typingStates
+      .filter(state => Object.values(state)[0]) // Keep only active users (true)
+      .map(state => Object.keys(state)[0]), // Extract the username
+  )
```

Comment:
The use of a `SvelteMap` with only currently typing users would eliminate the need for this

Also, we should prefer user IDs and not usernames for this (especially once display names are added in the future), for that purpose, the user IDs of people typing should be considered in the `$effect` below